### DISCUSSION TITLE
Possibility to link transfer to a bill

### DISCUSSION THREAD

**Discussion by vladonemo** [firefly-iii/firefly-iii#6824] [Firefly III]
I can see in the code that linking a transaction to a bill is only possible to withdrawals. Is there any reason why not to allow transfers to be linked to a bill?

My use-case is very simple. I use the "bills" functionality to keep track of (reoccurring) monthly expenses. This includes investments and savings.

**Comment by JC5** [comment]
It's done to keep things simple and track the payment of bills through withdrawals.

**Reply by vladonemo** [reply]
Thanks for the reply. In fact, I see it as more complicated now. Not only the code is complicated (the check on transaction type is on multiple places) but it restricts the user from doing their workflow. It is also not very intuitive -> for instance, you can create an automatic rule to assign category, budget and bill to a transaction but it doesn't happen (the check on withdrawal type is also in the rule source code). Moreover, I can link a transaction to a bill directly in the DB and it all works OK in the UI, meaning, the check on transaction type is missing in the bills overview ;)

**Reply by BackedUpBooty** [reply]
I'd also like to see this as an option. My specific use-case: I have a Credit Card created as an Asset account. This asset account has withdrawals against it. The credit card is then paid off by one of my current account which is another Asset account. This is seen as a 'transfer' in the system, but is in effect a payment/expense as well. 

The only way I can accurately track this *and* have it linked to a bill is by doing two separate transactions, one as a withdrawal/expense which has all the rules on it, and another as a separate sourceless transaction in transfers. It balances out but isn't particularly intuitive or graceful. Is there some other way I'm missing that could work here? Or am I using this in a way that wasn't anticipated maybe?

**Reply by JC5** [reply]
> This is seen as a 'transfer' in the system, but is in effect a payment/expense as well.

I don't consider this to be a payment / expense. The expenses were on the CC, now you're just balancing things.

**Reply by BackedUpBooty** [reply]
Except the credit card IS a bill, it's an expense account in its own right too. Some people will pay off the whole thing each month, others won't and will pay somewhere between minimum and full, and then incur further interest charges on their next bill (which isn't technically an expense). Being able to track it like that would help.

**Reply by JC5** [reply]
So, then the interest is the expense. :)

**Reply by BackedUpBooty** [reply]
So another reason - expenses on the credit card asset account are counted as expenses. Treating the credit card payment as an expense (the only way to link it to a bill) is also counted as an expense. This means that anything paid for on a credit card is counted twice. This isn't ideal.

**Reply by BackedUpBooty** [reply]
It's a shame that this conversation has stopped. If there's a better way to track credit card expenses and then be able to show the payment to the credit card each month then I'd love to hear about it. Right now about 80% of my daily expenses go on a credit card, and that means that when I create another transaction to pay it off Firefly is thinking I've spent that 80% again in one go.

**Reply by Dekumori** [reply]
Apologies for the necro.  First I want to thank you for the tool.  With Mint shutting down, it's been both a fun project to set this up and a good replacement.

If I'm not mistaken, the Bills section is primarily a way to stay on top of your obligations that have due dates.  Expenses are primarily tracked under categories and budgets.  Even though a credit card payment is technically a transfer and not an expense, it is still an obligation with a due date, so I believe it should be trackable in the bills category.

This, of course, is not a demand for the feature.  You've done fantastic work and aren't obligated to make any changes for anyone else.

**Reply by JC5** [reply]
Personally, I don't track these transfers as a bill. I keep a piggy bank for the collected amount, based on the expenses on the credit card.

**Reply by Dekumori** [reply]
That makes sense.  Ideally, that's what I'd like to do, but I'm currently in quite a bit of credit card debt.  In my mind, the piggy bank option would be ideal if I had my finances under control and used my credit cards as transfer accounts.  Currently, I don't use my credit cards at all and they're essentially functioning as a loan, which makes tracking them as a bill more appealing.  I'll give the piggy bank solution more thought though.  I appreciate the suggestion.

**Reply by JC5** [reply]
That's fair, I had a similar challenge when I paid of other debts. If it's a big debt and you don't use the card, consider using a liability account, despite the documentation's recommendation not to do so. Transactions to a liability are withdrawals and can be linked to a bill.
